tools/pythonpkg/src/native/python_objects.cpp

```diff
@@ -232,43 +230,42 @@ Value PyTime::ToDuckValue() {
 	return Value::TIME(duckdb_time);
 }
 
-int32_t PyTime::GetHours(PyObject *obj) {
-	return PyDateTime_TIME_GET_HOUR(obj); // NOLINT
+int32_t PyTime::GetHours(py::handle &obj) {
+	return PyDateTime_TIME_GET_HOUR(obj.ptr()); // NOLINT
 }
 
-int32_t PyTime::GetMinutes(PyObject *obj) {
-	return PyDateTime_TIME_GET_MINUTE(obj); // NOLINT
+int32_t PyTime::GetMinutes(py::handle &obj) {
+	return PyDateTime_TIME_GET_MINUTE(obj.ptr()); // NOLINT
 }
 
-int32_t PyTime::GetSeconds(PyObject *obj) {
-	return PyDateTime_TIME_GET_SECOND(obj); // NOLINT
+int32_t PyTime::GetSeconds(py::handle &obj) {
+	return PyDateTime_TIME_GET_SECOND(obj.ptr()); // NOLINT
 }
 
-int32_t PyTime::GetMicros(PyObject *obj) {
-	return PyDateTime_TIME_GET_MICROSECOND(obj); // NOLINT
+int32_t PyTime::GetMicros(py::handle &obj) {
+	return PyDateTime_TIME_GET_MICROSECOND(obj.ptr()); // NOLINT
 }
 
-PyObject *PyTime::GetTZInfo(PyObject *obj) {
-	return PyDateTime_TIME_GET_TZINFO(obj); // NOLINT
+py::object PyTime::GetTZInfo(py::handle &obj) {
+	// The object returned is borrowed, there is no reference to steal
+	return py::reinterpret_borrow<py::object>(PyDateTime_TIME_GET_TZINFO(obj.ptr())); // NOLINT
```

Comment:
This could be a `py::handle` instead but this function is kind of weird, in that it returns a non-owning reference while it's safe to assume that other CPython functions that return a PyObject* return an owned reference.

When this is changed to a `reinterpret_steal` this causes a crash, because the reference is not owned.

So I've opted to explicitly return a borrowed py::object here instead of a py::handle